### PR TITLE
NDEV-1966: Rewrite tests

### PR DESCRIPTION
**Pull request by romanova-natasha**

### CODE REVIEW

**Comment by romanova-natasha** [comment]
Tests run are failed cause it reffers to latest neon-tests image tag. But test cases are not merged to develop branch yet. To check if tests are run successfully please see manually run pipeline with correct test branch https://github.com/neonlabsorg/tracer-api/actions/runs/6228834272

**Comment by Deniskore** [comment]
In order to keep the history as it should be, you should rebase commits, get rid of many fix messages in commits (squash), all the remaining commits should be linked to the task number, for example
```NDEV-1966: Useful message```
Thanks in advance.

**Review by Deniskore — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
   workflow_dispatch:
+    inputs:
+        neon_tests_image:
+          description: 'Neon tests image tag'
+          required: true
+          default: 'neonlabsorg/neon_tests:latest'
+          type: string
 
 env:
   NEON_REVISION: 2c02d4a56a4509a17493e189a410d689cca805c8
```
Comment (gigimon) [inline .github/workflows/ci.yaml:14]: useless or?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -110,37 +115,73 @@ jobs:
           tags: neonlabsorg/neon-rpc:${{env.BUILDKITE_COMMIT}}
           labels: neonlabsorg/neon-rpc:${{env.BUILDKITE_COMMIT}}
 
-      - name: Docker-compose pull
-        run: NETWORK_NAME=net_${GITHUB_SHA} docker-compose -f docker-compose-test.yml pull
+  tests:
+    needs: [check, fmt, clippy, test-devnet, build-neon-tracer-docker-image, build-neon-rpc-docker-image]
```
Comment (gigimon) [inline .github/workflows/ci.yaml:119]: you could simplify depends and use only build-* actions

**Review by gigimon — APPROVED** [review]

**Pull request merged**